### ISSUE TITLE
[Bug] Lightweight Wallet bugs/issues on wallet with very large transaction history

### ISSUE THREAD

**Issue by paspy** [Chia-Network/chia-blockchain#11526]
### What happened?

Someone/people is/are constantly attacking the network (dust storm) and spamming our farming address. It causes our wallet contains a very large transaction history. (Still feeding us 1 or 2 mojos every second at the time I submit this report)
![image](https://user-images.githubusercontent.com/1343488/168411791-60ad8c44-7ad3-4134-821a-92e9d431fb7b.png)

Apparently, the lightweight wallet is not able to handle such large transactions correctly. I have tried to use a local fully synced node to sync a fresh start lightweight wallet. The total time is about 5 days and when it is fully synced the balance is way off compared with the original full node wallet (chia version 1.2.11)

This issue/bug prevents me to upgrade the farming rigs to the latest chia version because the wallet cannot fully sync correctly.

Is there any way to split the 1.2.11 full node wallet to work with the latest Chia node so it can bypass the issue? For us this particular issue, the lightweight wallet takes much more time to sync than the full node wallet. (and the balance is not correct)

### Version

1.3.4 and 1.3.5

### What platform are you using?

Linux

### What ui mode are you using?

CLI

**Comment by mariano54** [comment]
Also, did you make sure to keep it online for a couple of hours after the sync finished? Sometimes is takes a while to find some of the missing coins

**Comment by zhangyongchang88** [comment]
我和你一样遇到了丢失xch和沙尘暴攻击的问题，我很早也跟他们反应，好像无人理会，可笑的是，我再1.3.5基础上重新安装1.2.11同步完成后，所有硬币消失了。我不得不安装回1.3.5
I have the same problem as you with lost xch and sandstorm attack, I also reacted to them very early, it seems that no one cares, the funny thing is, after I reinstalled 1.2.11 on the basis of 1.3.5, all the coins disappeared . I had to install back to 1.3.5

**Comment by zhangyongchang88** [comment]
我很早就反应过轻量级钱包的问题，新版本应该是可选轻量级钱包模式，或者最初的那种模式。应该让用户可选，所谓的升级后数据库使用减少了，仅仅是本地不存储钱包数据库了。太可笑了。
I reacted to the problem of lightweight wallets very early, and the new version should be an optional lightweight wallet mode, or the original mode. It should be optional for users. The so-called database usage after the upgrade is reduced, but the wallet database is not stored locally. That's ridiculous.

**Comment by mariano54** [comment]
We are working on fixing the issue, we would appreciate as much information as possible

**Comment by zhangyongchang88** [comment]
> x 版本更
1.3.5 The problem still exists

**Comment by zhangyongchang88** [comment]
> 请确认在您的本地上测试过此操作。 1.3.5 和之前的1.3.5 比之前的 1.3.x 版本更熟练查找所有内部。
> 
> 感谢您的更新。

1.3.5 The problem still exists

**Comment by zhangyongchang88** [comment]
> 另外，您是否确保在同步完成后保持在线几个小时？有时需要一段时间才能找到一些丢失的硬币

My current computer wallet has been open for more than 2 days, but the amount is not correct. I'm having trouble with losing 8 xch recently. This is a serious blockchain security incident.
我现在的电脑钱包就已经开启超过2天了，可是数量就不正确。我正在因为最近丢了8个xch而烦恼。这是一个严重的区块链安全事故。

**Comment by paspy** [comment]
> We are working on fixing the issue, we would appreciate as much information as possible

I have tested a few ways that might cause the missing coins from two different wallets (the 1.2.11 full node wallet and the 1.3.5 lightweight wallet)

It seems like these two types of wallets can't always find all the coins from their addresses.

For example, suppose the full node wallet has 10 XCH, and the lightweight wallet is only searchable 2 XCH from the same private key. They both have a different First Address (the default wallet address)

I tried to generate a bunch of different addresses from both these two wallets and send XCH to these two different lists of addresses and the result is: Both the full node wallet and the lightweight wallet are not always able to show the correct balance., it means there is only a few addresses are overlapping from these wallet's point of view. 

Hope this info helps
